tests/freqtradebot/test_integration.py

```diff
@@ -478,9 +479,24 @@ def test_dca_order_adjust(default_conf_usdt, ticker_usdt, leverage, fee, mocker)
     assert pytest.approx(trade.amount) == 91.689215 * leverage
     assert pytest.approx(trade.orders[-1].amount) == 91.689215 * leverage
     assert freqtrade.strategy.adjust_entry_price.call_count == 0
+
+    print("BEFORE Process trade.orders")
+    print(trade.orders)
+
+    # adding this will prevent the second exit order creation, but this test case must be handled properly
+    #freqtrade.strategy.custom_exit = MagicMock(return_value=None)
+
+    caplog.clear()
+    caplog.set_level(logging.DEBUG)
     # Process again, should not adjust entry price
     freqtrade.process()
```

Comment:
I see that a second exit order is created after the first expected one. 

    This is because there is a small amount of the position that remains untied. So the "if trade.has_untied_assets" condition let the bot continue to exit.
    The exit_positions handling will create another exit order with the full registered trade.amount as the new exit order amount, instead of the real remaining untied amount.

    It could be necessary to replace the trade.amount value by a result based on all orders of the trade. What do you think about this?

Reply:
I saw that `recalc_trade_from_orders` is doing the update between the trade and related orders. Maybe we just need to make sure that in `execute_trade_exit`  `trade._safe_exit_amount` return remaining untied_assets only.

Reply:
@xmatthias what do you think about that?

Reply:
it's unclear to me what "untied assets" are supposed to be - it seems to be mostly a copy of what `recalc_trade_from_orders` does - but in a "i do it again and again (and again and again and again) mode" (being a property and all) - ignoring that potentially most orders never even filled a tiny bit.

in my opinion - each trade has a quantity - `.amount` - which corresponds to the amount you currently got.

The way you describe it most likely means that the (in my opinion pointless) newly introduced property has calculation flaws (i suspect it ignores fees - but i didn't look closely at that).

Reply:
The goal of `trade.has_untied_assets` was to check if the contracts/crypto we hold, have not been already tied to an existing open exit order.

Reply:
well we don't expect to have more than one open order at any point in time for each trade (excluding stop orders).

(at least not in the scope of this PR).